### PR TITLE
Guided Dinovol / MedNeXt stack and autoregressive mesh MVP

### PR DESCRIPTION
**Pull request by giorgioangel**
## Summary
This PR brings three major lines of work onto `main`:

1. guided Dinovol-based segmentation / guidance infrastructure
2. MedNeXt architecture support and benchmarks
3. a first end-to-end `autoreg_mesh` prototype for tifxyz mesh continuation, including training, inference, geometry losses, logging, and experimental debiasing configs

The branch is intentionally broad because `autoreg_mesh` was built on top of the same long-lived branch that also introduced the guided Dinovol / MedNeXt stack.

### Guided Dinovol / MedNeXt
- Added local Dinovol backbone loading and guided feature-conditioning infrastructure.
- Added multiple guided-network modes, benchmarks, config manager updates, trainer/logging support, and guidance-focused tests.
- Added MedNeXt v1 / v2 model support, wrappers, configs, and benchmark coverage.
- Added pretrained DINO / pixelshuffle-related decoder support and associated config/test coverage.

### `autoreg_mesh` MVP
- Added a self-contained package under `vesuvius.neural_tracing.autoreg_mesh` with:
  - serialization utilities for frontier-ordered wrap continuation
  - dataset wrapper and collate path
  - autoregressive decoder model
  - losses
  - inference path
  - local trainer
  - benchmark entrypoint
  - README / usage notes
- Added configs for `0139` 2um training runs, including joint-pointer and axis-factorized variants.
- Added stored-surface sampling support while keeping full-resolution serialization as the main path.

### Training / inference / geometry work in `autoreg_mesh`
- W&B-aware local training loop with checkpointing, validation, rollout metrics, and image logging.
- Projection previews plus XY slice overlays of mesh lines on volume slices.
- Scheduled sampling, delayed offset supervision, direct soft-xyz supervision, seam supervision, local no-flip barrier, geometry metric loss, Symmetric Dirichlet loss, and rollout validation metrics.
- Axis-factorized coarse prediction mode as a config-selectable alternative to the joint pointer head.
- Optional INSID3-style frozen-feature debiasing ablation on the Dinovol token stream, with dedicated debias16 configs.

### Tests / docs
- Added extensive smoke and regression coverage for:
  - guided model construction / trainer behavior
  - config compatibility
  - autoreg mesh serialization / forward pass / training / inference / logging / geometry losses / benchmark paths
- Added docs for guided Dinovol and the local autoreg mesh package.

### `autoreg_mesh`
- Augmentations are **not wired in the current autoreg mesh training path**.
  - The dataset wrapper currently runs with `apply_augmentation=False` and `apply_perturbation=False`.
  - Volume-only safe augmentations still need to be intentionally integrated rather than inherited implicitly from `EdtSegDataset`.
- No global non-adjacent self-collision penalty or constrained decoding is implemented yet.
  - Current anti-self-intersection protection is local (triangle orientation / no-flip barrier).
- The INSID3-style conditioning debiasing is implemented as an **optional ablation**, not a default.
  - It still needs long-run training evidence before deciding whether it helps.
- `autoreg_mesh` remains on its own local trainer path.
  - It is not integrated into the generic trainer stack / distributed training flow.
- No coarse-to-fine refinement stage or tokenized strip/block representation yet.
  - This is still a direct per-vertex MVP.

### Validation / productionization
- This branch adds targeted smoke/regression tests, but full production-scale experiment validation is still pending.
- The new debias16 configs were added, but branch-level review should treat them as experiment configs rather than settled defaults.

## Testing
Targeted local suites exercised during development include:

```bash
uv run --project vesuvius --extra models --extra tests python -m pytest vesuvius/tests/neural_tracing/test_autoreg_mesh_serialization.py vesuvius/tests/neural_tracing/test_autoreg_mesh_model_smoke.py
```

The branch also adds and updates model/build/trainer/config tests for the guided Dinovol / MedNeXt stack.

## Review Notes
- This PR is large because the branch diverged significantly from `main` while building `autoreg_mesh`.
- Reviewers may find it easiest to inspect by subsystem:
  - guided Dinovol / MedNeXt infrastructure
  - `autoreg_mesh` package and configs
  - test additions
- The most explicitly experimental pieces in this PR are:
  - axis-factorized coarse mode
  - INSID3-style feature debiasing
  - debias-specific training configs

### CODE REVIEW

**Comment by giorgioangel** [comment]
Suggested tighter PR framing focused on `autoreg_mesh`:

**Proposed title**
`Autoregressive tifxyz mesh completion MVP with geometry-aware training and Dinovol conditioning`

**Proposed body**
## Summary
This PR adds a first end-to-end `vesuvius.neural_tracing.autoreg_mesh` prototype for autoregressive tifxyz mesh completion from a 3D volume crop and a partial wrap prompt.

The implementation includes:
- frontier-ordered tifxyz serialization and reconstruction
- dataset / collate path for one-wrap-per-example split conditioning
- frozen Dinovol-conditioned autoregressive decoder
- joint-pointer and axis-factorized coarse prediction modes
- local training loop with W&B logging, checkpointing, and rollout validation
- inference / lattice reconstruction utilities
- geometry-aware training losses and visualization helpers
- experiment configs for `0139` 2um runs, including SD, axis-factorized, and debias variants

## Implemented
### Core MVP
- Added a self-contained `autoreg_mesh` package with:
  - serialization
  - dataset wrapper
  - model
  - losses
  - inference
  - train / benchmark entrypoints
  - README
- Default representation remains per-vertex autoregression over tifxyz quad lattices, not generic mesh face generation.

### Conditioning and decoding
- Frozen Dinovol memory conditioning.
- Shared Dinovol-style 3D RoPE integrated into decoder self-attention and cross-attention.
- Config-selectable coarse head:
  - `joint_pointer`
  - `axis_factorized`
- Optional INSID3-style frozen-feature debiasing ablation on the Dinovol token stream.

### Training and geometry supervision
- W&B-aware local trainer with checkpoint resume and image logging.
- Scheduled sampling with strip-wise replacement and feedback gating.
- Direct soft-xyz supervision.
- Seam supervision across the conditioning / continuation frontier.
- Local no-flip barrier.
- Geometry metric loss and Symmetric Dirichlet regularization.
- Rollout validation metrics and XY slice mesh overlays on volume slices.

### Tests
- Added smoke and regression tests for:
  - serialization order and reconstruction
  - forward/loss shape checks
  - inference reconstruction
  - scheduled sampling
  - geometry losses
  - W&B logging
  - benchmark path
  - optional feature debiasing

## Still Missing / Follow-up
- Augmentations are **not wired in the current `autoreg_mesh` training path**.
  - The dataset wrapper currently runs with `apply_augmentation=False` and `apply_perturbation=False`.
  - Safe volume-only augmentations still need explicit integration.
- No global non-adjacent self-collision penalty yet.
  - Current self-intersection protection is local (triangle orientation / no-flip barrier).
- No constrained decoding / rejection sampling at inference yet.
- No coarse-to-fine refinement stage or learned strip/block tokenization yet.
- The INSID3-style feature debiasing is implemented only as an optional ablation; it is not recommended as a default without training evidence.
- `autoreg_mesh` still uses its own local trainer path and is not integrated into the generic distributed trainer stack.

## Testing
Targeted local testing during development included:

```bash
uv run --project vesuvius --extra models --extra tests python -m pytest vesuvius/tests/neural_tracing/test_autoreg_mesh_serialization.py vesuvius/tests/neural_tracing/test_autoreg_mesh_model_smoke.py
```

## Note on Branch Scope
This branch also contains broader guided Dinovol / MedNeXt infrastructure changes relative to `main`. The `autoreg_mesh` work is the main deliverable, but reviewers should be aware the branch is not isolated to that package alone.

If you want, I can also tighten this further into a shorter reviewer-facing version.

**Comment by giorgioangel** [comment]
@codex review

**Comment by giorgioangel** [comment]
@codex review

**Comment by giorgioangel** [comment]
@codex review

**Comment by giorgioangel** [comment]
@codex review

**Comment by giorgioangel** [comment]
@codex review

**Comment by giorgioangel** [comment]
@codex review
